modules/auxiliary/gather/chrome_debugger.rb

```diff
@@ -0,0 +1,124 @@
+class MetasploitModule < Msf::Auxiliary
+  require 'eventmachine'
+  require 'faye/websocket'
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Chrome Debugger Arbitrary File Read / Arbitrary Web Request',
+      'Description' => %q{
+        This module uses the Chrome Debugger's API to read
+        files off the remote file system, or to make web requests
+        from a remote machine.  Useful for cloud metadata endpoints!
+      },
+      'License' => MSF_LICENSE,
+      'Author' => [
+        'Adam Baldwin (Evilpacket)', # Original ideas, research, proof of concept, and msf module
+        'Nicholas Starke (The King Pig Demon)' # msf module
+      ],
+      'Privileged' => false,
+      'Targets' => [
+      ],
+      'DisclosureDate' => 'Sep 24 2019',
+      'DefaultOptions' => {
+      },
+      'DefaultTarget' => 0
+    ))
+
+    register_options(
+      [
+        Opt::RHOST,
+        Opt::RPORT(9222),
+        OptString.new('FilePath', [ false, 'File to fetch from remote machine.']),
+        OptString.new('Url', [ false, 'Url to fetch from remote machine.']),
+        OptInt.new('Timeout', [ true, 'Time to wait for response', 10])
+      ]
+    )
+
+    deregister_options('Proxies')
```

Comment:
```suggestion
    deregister_options('Proxies')
    deregister_options('VHOST')
```

As far as I can tell `VHOST` which is provided by the `mixin` is not needed in this module.

Reply:
Looks like `SSL` could also be deregistered as not used.

```
msf5 auxiliary(gather/chrome_debugger) > options

Module options (auxiliary/gather/chrome_debugger):

   Name      Current Setting     Required  Description
   ----      ---------------     --------  -----------
   FilePath                      no        File to fetch from remote machine.
   RHOSTS    192.168.17.102      yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT     9222                yes       The target port (TCP)
   SSL       false               no        Negotiate SSL/TLS for outgoing connections
   Timeout   10                  yes       Time to wait for response
   Url       https://github.com  no        Url to fetch from remote machine.
```